src/backend/apps/webcam/tasks.py

```diff
@@ -139,3 +150,71 @@ def update_webcam_image(webcam):
 
     except Exception as e:
         logger.error(e)
+
+
+# Camera ordering tasks
+def add_order_to_camera_group(key, cams):
+    # Only process if reference linestring exists
+    route_geometry = RouteGeometry.objects.filter(id=key).first()
+    if route_geometry:
+        # Combine all coordinates into a single linestring for sorting
+        all_coords = ()
+        for coord in route_geometry.routes.coords:
+            all_coords += coord
+
+        # Save distance along route to cams
+        route_ls = LineString(all_coords)
+        for cam in cams:
+            cam.route_distance = route_ls.project(cam.location)
+
+        # Sort cams by distance along route and update
+        last_point = None
+        current_order = 0
+        for sorted_cam in sorted(cams, key=lambda c: (c.route_distance, c.name)):
+            # Increase order unless same location as last cam
+            if last_point and not sorted_cam.location.equals(last_point):
+                current_order += 1
+
+            # Use update here to avoid saving the entire object
+            Webcam.objects.filter(id=sorted_cam.id).update(route_order=current_order)
+
+            # Update last cam
+            last_point = sorted_cam.location
+
+
+def add_order_to_cameras():
+    # Sort before grouping by route name
+    query = Webcam.objects.all().order_by('highway', 'highway_description')
+
+    # Group by route name and add order to cams in each group
+    for route_name, cam_group in groupby(query, lambda x: x.highway if x.highway != '0' else x.highway_description):
+        add_order_to_camera_group(route_name, list(cam_group))
+
+    # Rebuild cache
+    WebcamAPI().set_list_data()
+
+
+def build_route_geometries():
```

Comment:
Overall, we need an explanation in code of why we're doing this (meaning getting the route, storing it, and then using it to order cameras).  A docstring for this function would be a good place to outline the intended purpose (along with the JIRA ticket number for the story); the highway coords structure should then be commented to say "used by build_route_geometries".

Reply:
Updated docstrings and comments via amend commit.